### PR TITLE
PAY-3692 - Azure rm provider upgrade changes

### PR DESCRIPTION
**Pull request by amitchidrewar1301**
Version upgrade to 2.x azure rm

### CODE REVIEW

**Review by vijayrajagopalan-hmcts — COMMENTED** [review]
Remove string interpolation syntax where applicable. For example "${var.component}-POSTGRES-PASS" can be changed to **join("-", [var.component, "POSTGRES-PASS"])**

**Review by naikajah — APPROVED** [review]

**Comment by amitchidrewar1301** [comment]
> Remove string interpolation syntax where applicable. For example "${var.component}-POSTGRES-PASS" can be changed to **join("-", [var.component, "POSTGRES-PASS"])**

Suggested changes are done now for interpolation syntax

**Review by vijayrajagopalan-hmcts — APPROVED** [review]

File: config/owasp/suppressions.xml [hunk 1]
```diff
@@ -23,5 +23,6 @@
     <cve>CVE-2020-13935</cve>
     <cve>CVE-2020-8022</cve>
     <cve>CVE-2017-18640</cve>
+    <cve>CVE-2020-5421</cve>
```
Comment (AyanDuttaHMCTS) [inline config/owasp/suppressions.xml:26]: Please add a note mentioning the reason for suppressing the CVE

Comment (amitchidrewar1301) [reply config/owasp/suppressions.xml:26]: The above CVE is coming for spring modules with version 5.2.4 i.e spring - aop, spring -aspects, spring, etc. These are transactive dependencies of spring boot version 2.2.5.  Spring boot version upgrade needs to happen to resolve these CVEs.

Comment (AyanDuttaHMCTS) [reply config/owasp/suppressions.xml:26]: Makes sense, thanks @amitchidrewar1301

**Review by AyanDuttaHMCTS — APPROVED** [review]

**Pull request merged**